apm-agent-plugins/apm-profiling-plugin/src/test/java/co/elastic/apm/agent/profiler/CallTreeTest.java

```diff
@@ -262,6 +265,30 @@ void testDectivationBeforeEnd() throws Exception {
         });
     }
 
+    /*
+     * [1            ]    [1            ]
+     *  [a          ]      [a          ]
+     *   [2   ]  [3]        [b    ] [3]   <- b is supposed to stealChildIdsFom(a)
+     *    [b   ]            [2   ]           however, it should only steal 2, not 3
+     */
+    @Test
+    void testDectivationBeforeEnd2() throws Exception {
+        assertCallTree(new String[]{
+            "   bbbb b      ",
+            " a aaaa aa a a ",
+            "1 2    2  3 3 1"
```

Comment:
```suggestion
            "   bbbb b      ",
            " a aaaa a a a ",
            "1 2    2 3 3 1"
```
Just to make sure that one measurement is enough

Reply:
Hm, that doesn't work because `b` is still considered the top of the stack at the time `3` gets activated.

Reply:
Why isn't that enough, if you know that activation of `3` happens after deactivation of `2` and you know that `b` cannot be a child of `3`?

Reply:
Because of situations like these:

```
 aa  aa 
1  22  1
```

We have to add each activation to the `childIds` of the current top of the stack.

However, it least to errors in situations like these:
```
 bbb      
 aaa aa a 
1   2  2 1
```

As 2 is falsely declared a child of b when it's actually a child of a. Currently not sure how to fix it without introducing regressions.

Reply:
I think I've managed to fix it